qt/scientific_interfaces/ISISReflectometry/Reduction/Row.h

```diff
@@ -48,10 +50,19 @@ class MANTIDQT_ISISREFLECTOMETRY_DLL Row : public Item {
   bool hasOutputWorkspace(std::string const &wsName) const;
   void renameOutputWorkspace(std::string const &oldName, std::string const &newName) override;
 
+  void setParent(IGroup *parent) const;
```

Comment:
This shouldn't be marked const, and we should drop the mutable on `m_parent`

As a rule of thumb observable changes shouldn't have const on them, if this was an internal detail we can't see (e.g. a cache) then `const` and mutable would be fine

Otherwise people can run into undefined behaviour or bugs when they grab a `const IGroup` that changes underneath them

Reply:
We had a lot of issues making this non-const (the whole GUI is essentially built around Rows being const!). The parent pointer is the only part of a row that can change after a row is created. I think that's just for annoying internal reasons like if you move a group its pointer changes so the row's parent needs to be updated. There might be a way around this but I think it's out of scope for this PR which has quite a lot of refactoring in it already!